File: Cpp/Problem_1614.cpp
#include <iostream>
#include <string>

using namespace std;

// Problem - 1614: Maximum Nesting Depth of the Parentheses
// C++ Solution!
class Solution {
public:
    int maxDepth(string s) {
        int res = 0;
        int curDepth = 0;

        for (char i : s) {
            if (i == '(')
                curDepth++;
            else if (i == ')')
                curDepth--;

            res = max(res, curDepth);
        }

        return res;
    }
};

int main() {
    Solution sol;

    cout << sol.maxDepth("(1+(2*3)+((8)/4))+1") << endl;
    cout << sol.maxDepth("(1)+((2))+(((3)))") << endl;

    return 0;
}

File: Python/Problem_1614.py
# Problem - 1614: Maximum Nesting Depth of the Parentheses
# Python3 Solution!
class Solution:
    def maxDepth(self, s: str) -> int:
        res, curDepth = 0, 0

        for i in s:
            if i == '(':
                curDepth+=1
            elif i == ')':
                curDepth-=1

            res = max(res, curDepth)

        return res

if __name__=="__main__":
    sol = Solution()

    print(sol.maxDepth("(1+(2*3)+((8)/4))+1"))
    print(sol.maxDepth("(1)+((2))+(((3)))"))